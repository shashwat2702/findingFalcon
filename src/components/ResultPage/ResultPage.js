/* eslint-disable camelcase */
/* eslint-disable react/forbid-prop-types */
import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import './ResultPage.scss';
import Button from '../shared/Button/Button';

class ResultPage extends Component {
  constructor(props) {
    super(props);
    const { location, history } = props;
    const { state } = location;
    let foundStatus = '';
    let totalTimeTaken = 0;
    let foundOnPlanet = '';
    if (!state) {
      history.push('/');
    } else {
      const { key, timeTaken } = state;
      const { status, planet_name } = key;
      foundStatus = status;
      totalTimeTaken = timeTaken;
      foundOnPlanet = planet_name;
    }
    this.state = {
      foundStatus,
      totalTimeTaken,
      foundOnPlanet,
    };
  }

  onSubmit = () => {
    const { history } = this.props;
    history.push('/');
  }

  render() {
    const { foundStatus, totalTimeTaken, foundOnPlanet } = this.state;
    console.log(this.props);
    return (
      <div>
        <h1>Finding Falcon! </h1>
        {(foundStatus === 'success')
          ? <h2>Success! Congratulations on Finding Falcone. King Shan is mighty pleased.</h2>
          : <h2>Sorry! Falcone was not found on selected Planets. King Shan is dissappointed.</h2>
        }
        {
            (foundStatus === 'success') && (
            <div>
              <h2>
                Time taken
                {' '}
                {totalTimeTaken}
              </h2>
              <h2>
                Planet Found:
                {' '}
                {foundOnPlanet}
              </h2>
            </div>
            )
        }
        <div className="startAgainButton">
          <Button
            label="Start Again"
            onClick={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}
ResultPage.propTypes = {
  location: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
};
export default withRouter(ResultPage);
