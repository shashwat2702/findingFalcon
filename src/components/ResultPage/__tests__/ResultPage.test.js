import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { ResultPage } from '../ResultPage';

jest.mock('react-router-dom');

describe('ResultPage snapshot', () => {
  it('should render correctly when Falcon is found', () => {
    const location = {
      state: {
        timeTaken: 200,
        key: {
          status: 'success',
          planet_name: 'DonLon',
        },
      },
    };
    const history = {
      push: jest.fn(),
    };
    const wrapper = renderer.create(<ResultPage history={history} location={location} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly when Falcon is not found', () => {
    const location = {
      state: {
        timeTaken: 200,
        key: {
          status: 'failure',
          planet_name: 'DonLon',
        },
      },
    };
    const history = {
      push: jest.fn(),
    };
    const wrapper = renderer.create(<ResultPage history={history} location={location} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('constructor', () => {
  it('call history.push with / when data is not received', () => {
    const location = {
      state: false,
    };
    const history = {
      push: jest.fn(),
    };
    const wrapper = shallow(<ResultPage history={history} location={location} />);
    const instance = wrapper.instance();
    expect(history.push).toHaveBeenCalledWith('/');
    expect(instance.state.foundStatus).toEqual('');
  });
});

describe('onSubmit', () => {
  it('call history.push with /', () => {
    const location = {
      state: {
        timeTaken: 200,
        key: {
          status: 'failure',
          planet_name: 'DonLon',
        },
      },
    };
    const history = {
      push: jest.fn(),
    };
    const wrapper = shallow(<ResultPage history={history} location={location} />);
    const instance = wrapper.instance();
    instance.onSubmit();
    expect(history.push).toHaveBeenCalledTimes(1);
    expect(history.push).toHaveBeenCalledWith('/');
  });
});
