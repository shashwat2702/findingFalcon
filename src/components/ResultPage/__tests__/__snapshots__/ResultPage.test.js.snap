// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultPage snapshot should render correctly when Falcon is found 1`] = `
<div>
  <h1>
    Finding Falcon! 
  </h1>
  <h2>
    Success! Congratulations on Finding Falcone. King Shan is mighty pleased.
  </h2>
  <div>
    <h2>
      Time taken
       
      200
    </h2>
    <h2>
      Planet Found:
       
      DonLon
    </h2>
  </div>
  <div
    className="startAgainButton"
  >
    <button
      className="editTaskButton"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Start Again
    </button>
  </div>
</div>
`;

exports[`ResultPage snapshot should render correctly when Falcon is not found 1`] = `
<div>
  <h1>
    Finding Falcon! 
  </h1>
  <h2>
    Sorry! Falcone was not found on selected Planets. King Shan is dissappointed.
  </h2>
  <div
    className="startAgainButton"
  >
    <button
      className="editTaskButton"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Start Again
    </button>
  </div>
</div>
`;
